data/shaders/dancenote.vert

```diff
@@ -1,19 +1,26 @@
 #version 330 core
 
-uniform mat4 projMatrix;
-uniform mat4 mvMatrix;
-uniform mat3 normalMatrix;
-uniform int noteType;
-uniform float hitAnim;
-uniform float clock;
-uniform float scale;
-uniform vec2 position;
-
 layout(location = 0) in vec3 vertPos;
 layout(location = 1) in vec2 vertTexCoord;
 layout(location = 2) in vec3 vertNormal;
 layout(location = 3) in vec4 vertColor;
 
+layout (std140) uniform shaderMatrices {
+	mat4 projMatrix;
+	mat4 mvMatrix;
+	mat4 normalMatrix;
+	mat4 colorMatrix;
+} shaderMatrices;
+
+layout (std140) uniform danceNote {
+	int noteType;
+	float hitAnim;
+	float clock;
+	float scale;
+	vec2 position;
```

Comment:
why do we use vec4's for nearly everything, but vec2's for dance? weren't dance notes supposed to be 3d models in the future?

Reply:
Don't know honestly.